ADVANCED.md

```diff
@@ -260,7 +238,80 @@ ReactDOM.render(
 );
 ```
 
-You can do this for [class components](https://gist.github.com/karol-majewski/befaf05af73c7cb3248b4e084ae5df71) too (Credit: [Karol Majewski](https://twitter.com/WrocTypeScript/status/1163234064343736326))
+The same using fat arrow function style:
+
+```tsx
+interface Props<T> {
+  items: T[];
+  renderItem: (item: T) => React.ReactNode;
+}
+
+// Note the `<T extends {}>` before the function definition. You can't use just `<T>` as it will confuse the TSX parser.
```

Comment:
As of now, the default inferred type for type parameters is `unknown`, not `{}`. To keep this behavior consistent when a generic function is defined in a `.tsx` file, the example could suggest doing

**Good**

```ts
<T extends unknown>
```

instead of 

**Bad**

```ts
<T extends {}>
```

**Also good**

Another solution is to [use a trailing comma inside the type parameter list](https://github.com/microsoft/TypeScript/issues/15713#issuecomment-499474386):

```ts
<T,>
```

Both will work equally well.